endpoint/app.py

```diff
@@ -0,0 +1,144 @@
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#-----------------------------------------------------------------------------
+
+import os
+import logging
+import logging.config
+import json
+
+from flask import Flask, request, Response
+import jwt
+import boto3
+from botocore.exceptions import ClientError
+
+application = Flask(__name__)
+
+
+logger = logging.getLogger(__name__)
+
+logging.config.dictConfig({
+    'version': 1,
+    'disable_existing_loggers': False,
+    'formatters': {
+        'simple': {
+            'format': '%(levelname)s: %(message)s',
+        },
+        'verbose': {
+            'format': '[%(asctime)s][%(module)s] %(levelname)s: %(message)s',
+        }
+    },
+    'handlers': {
+        'console': {
+            'level': 'DEBUG',
+            'class': 'logging.StreamHandler',
+            'formatter': 'verbose',
+        }
+    },
+    'loggers': {
+        '': {
+            'handlers': ['console'],
+            'level': 'DEBUG',
+            'propagate': False,
+        }
+    }
+})
+
+access = '80658bec0ce34880a9bbc3f3f50ca60b'
+secret = '9069592f9b314a52b26915e35fbef581'
+
+
+def add_object(access_key,secret_key, bucketname, prefix):
+    host = 'https://cf2.cloudferro.com:8080'
+    s3 = boto3.resource('s3',aws_access_key_id=access_key, aws_secret_access_key=secret_key, endpoint_url=host,)
+
+    bucket =s3.Bucket(bucketname)
+
+    bucket.put_object(Key=prefix)
+
+def upload_file(file_name, bucket, object_name=None):
+    """Upload a file to an S3 bucket
+
+    :param file_name: File to upload
+    :param bucket: Bucket to upload to
+    :param object_name: S3 object name. If not specified then file_name is used
+    :return: True if file was uploaded, else False
+    """
+
+    # If S3 object_name was not specified, use file_name
+    if object_name is None:
+        object_name = file_name
+    
+   
+    # Upload the file
+    host = 'https://cf2.cloudferro.com:8080'
+    s3_client = boto3.client('s3',aws_access_key_id='80658bec0ce34880a9bbc3f3f50ca60b', aws_secret_access_key='9069592f9b314a52b26915e35fbef581', endpoint_url=host,)
+    try:
+        response = s3_client.upload_file(file_name, bucket, object_name)
+    except ClientError as e:
+        logging.exception(e)
+        return False
+    return True
+
+def lookup_objects(access_key, secret_key, bucketname, pref):
+
+    host = 'https://cf2.cloudferro.com:8080'
+    s3 = boto3.resource('s3',aws_access_key_id=access_key, aws_secret_access_key=secret_key, endpoint_url=host,)
+    bucket=s3.Bucket(bucketname)
+
+    for obj in bucket.objects.filter():
+        if pref in obj.key:
+            return True
+        else: 
+            return False
+
+@application.route('/userinfo', methods=['GET'])
+def userinfo():
+
+    request.get_data()
+    encode_token = jwt.decode(request.headers['jwt-token'], verify=False, algorithms=['RS256'])
```

Comment:
I just did some research, and I think the [`Authorization` header with the `Bearer`](https://tools.ietf.org/html/rfc6750#section-2.1) scheme is appropriate here.

So it would look like this: `Authorization: Bearer <encoded-JWT>` so this would require some additional parsing.